airflow/ti_deps/deps/base_ti_dep.py

```diff
@@ -86,9 +87,6 @@ def get_dep_statuses(self, ti, session, dep_context=None):
         :param dep_context: the context for which this dependency should be evaluated for
         :type dep_context: DepContext
         """
-        # this avoids a circular dependency
-        from airflow.ti_deps.dep_context import DepContext
```

Comment:
Yes, we had cyclic import...
`Cyclic import (airflow.ti_deps.dep_context -> airflow.ti_deps.deps.dag_unpaused_dep -> airflow.ti_deps.deps.base_ti_dep) (cyclic-import)`